app/lib/katello/event_daemon/services/agent_event_receiver.rb

```diff
@@ -14,7 +14,7 @@ def initialize
 
           def handle(message)
             ::Katello::Util::Support.with_db_connection do
-              ::Katello::Agent::ClientMessageHandler.handle(message)
+              ::Katello::Agent::ClientMessageHandler.new(message).handle
```

Comment:
Love it. Was thinking along the same -> https://refactoring.com/catalog/replaceFunctionWithCommand.html